generate/connection.go

```diff
@@ -75,7 +84,14 @@ func findEdgeConnections(g *graph.Graph, n *graph.Node, visited map[string]struc
 					Direction: direc,
 				},
 			}
-			res = append(res, append(con, cons...))
+
+			// If there are no connections means it's an end Node,
+			// and as it is not a Node (it'll have entered the previous
+			// condition) it means it has no Node at the end so we do not
+			// return it
+			if len(cons) != 0 {
```

Comment:
we can remove the `if` conditions ? 

the append works even if `cons`'s length is 0

Reply:
Yeh but the logic is `res = append(res, append(con, cons...))` so it's to prevent to append the `con`.

I'll move the logic to be more clear and no initialize the `con` if not needed.